src/Sarif/NotYetAutoGenerated/Location.cs

```diff
@@ -0,0 +1,183 @@
+    /// A location within a programming artifact.
+    /// </summary>
+    [DataContract]
+    [GeneratedCode("Microsoft.Json.Schema.ToDotNet", "0.58.0.0")]
+    public partial class Location : PropertyBagHolder, ISarifNode
+    {
+        public static IEqualityComparer<Location> ValueComparer => LocationEqualityComparer.Instance;
+
+        public bool ValueEquals(Location other) => ValueComparer.Equals(this, other);
+        public int ValueGetHashCode() => ValueComparer.GetHashCode(this);
+
+        /// <summary>
+        /// Gets a value indicating the type of object implementing <see cref="ISarifNode" />.
+        /// </summary>
+        public SarifNodeKind SarifNodeKind
+        {
+            get
+            {
+                return SarifNodeKind.Location;
+            }
+        }
+
+        /// <summary>
+        /// Identifies the file and region.
+        /// </summary>
+        [DataMember(Name = "physicalLocation", IsRequired = false, EmitDefaultValue = false)]
+        public PhysicalLocation PhysicalLocation { get; set; }
+
+        /// <summary>
+        /// The human-readable fully qualified name of the logical location. If run.logicalLocations is present, this value matches a property name within that object, from which further information about the logical location can be obtained.
+        /// </summary>
+        [DataMember(Name = "fullyQualifiedLogicalName", IsRequired = false, EmitDefaultValue = false)]
+        public string FullyQualifiedLogicalName { get; set; }
+
+        /// <summary>
+        /// The index within the logicalLocations array that specifies the logical location associated with the result.
+        /// </summary>
+        [DataMember(Name = "logicalLocationIndex", IsRequired = false, EmitDefaultValue = false)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        [System.ComponentModel.DefaultValue(-1)]
+        public int LogicalLocationIndex { get; set; }
+
+        /// <summary>
+        /// A message relevant to the location.
+        /// </summary>
+        [DataMember(Name = "message", IsRequired = false, EmitDefaultValue = false)]
+        public Message Message { get; set; }
+
+        /// <summary>
+        /// A set of regions relevant to the location.
+        /// </summary>
+        [DataMember(Name = "annotations", IsRequired = false, EmitDefaultValue = false)]
+        public IList<Region> Annotations { get; set; }
+
+        /// <summary>
+        /// Key/value pairs that provide additional information about the location.
+        /// </summary>
+        [DataMember(Name = "properties", IsRequired = false, EmitDefaultValue = false)]
+        internal override IDictionary<string, SerializedPropertyInfo> Properties { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Location" /> class.
+        /// </summary>
+        public Location()
+        {
+            LogicalLocationIndex = -1;
```

Comment:
>LogicalLocationIndex =  [](start = 12, length = 23)

It is helpful for constructors to populate default values. JSON.NET won't do so except when deserializing. our current jschema code does not do this.

Reply:
Documented gap in the following issue:
[https://github.com/Microsoft/jschema/issues/57](https://github.com/Microsoft/jschema/issues/57)

---
In reply to: [243699926](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243699926) [](ancestors = 243699926)